Not working with iOS 15+

Hey, I'm sorry to disturb you all, I wanted to use this client in my swift app that's being developed and I wasn't successful in adding the library to the project dependencies - when I add is through this github, it shows about 17 errors when building, most of them talking about a method not being available in iOS lower than 13 or 15, even though the target iOS version is 15, am I doing something wrong or does anyone have an idea what could I try?

thanks in advance

I haven't found a way to upload a screenshot here so I copy pasted a build log containing only errors, since the normal one was long, hopefully it'll help


```
Showing All Errors Only

Build target Replicate with configuration Debug

SwiftEmitModule normal arm64 Emitting\ module\ for\ Replicate (in target 'Replicate' from project 'Replicate')
    cd /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift
    builtin-swiftTaskExecution -- /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -emit-module -experimental-skip-non-inlinable-function-bodies-without-types /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Client.swift /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Error.swift /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Extensions/Data+uriEncoded.swift /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Model.swift /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift -target arm64-apple-ios11.0-simulator -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk -I /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Products/Debug-iphonesimulator -I /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/lib -F /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Products/Debug-iphonesimulator/PackageFrameworks -F /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Products/Debug-iphonesimulator -F /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks -F /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.1.sdk/Developer/Library/Frameworks -no-color-diagnostics -enable-testing -g -module-cache-path /Users/******/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -swift-version 5 -enforce-exclusivity\=checked -Onone -D SWIFT_PACKAGE -D DEBUG -D Xcode -serialize-debugging-options -empty-abi-descriptor -Xcc -working-directory -Xcc /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -I/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/swift-overrides.hmap -Xcc -I/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Products/Debug-iphonesimulator/include -Xcc -I/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/DerivedSources-normal/arm64 -Xcc -I/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/DerivedSources/arm64 -Xcc -I/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/DerivedSources -Xcc -DSWIFT_PACKAGE -Xcc -DDEBUG\=1 -module-name Replicate -disable-clang-spi -target-sdk-version 16.1 -emit-module-doc-path /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/Objects-normal/arm64/Replicate.swiftdoc -emit-module-source-info-path /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/Objects-normal/arm64/Replicate.swiftsourceinfo -emit-objc-header-path /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/Objects-normal/arm64/Replicate-Swift.h -serialize-diagnostics-path /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/Objects-normal/arm64/Replicate-master-emit-module.dia -emit-dependencies-path /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/Objects-normal/arm64/Replicate-master-emit-module.d -o /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/Objects-normal/arm64/Replicate.swiftmodule -emit-abi-descriptor-path /Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/Build/Intermediates.noindex/Replicate.build/Debug-iphonesimulator/Replicate.build/Objects-normal/arm64/Replicate.abi.json

/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift:120:19: error: 'TaskPriority' is only available in iOS 13.0 or newer
        priority: TaskPriority? = nil,
                  ^
/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift:117:26: note: add @available attribute to enclosing instance method
    public mutating func wait(
                         ^
/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift:11:15: note: add @available attribute to enclosing generic struct
public struct Prediction<Input, Output>: Identifiable where Input: Codable, Output: Codable {
              ^
/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift:137:19: error: 'TaskPriority' is only available in iOS 13.0 or newer
        priority: TaskPriority?,
                  ^
/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift:133:25: note: add @available attribute to enclosing static method
    private static func wait(
                        ^
/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift:11:15: note: add @available attribute to enclosing generic struct
public struct Prediction<Input, Output>: Identifiable where Input: Codable, Output: Codable {
              ^

/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift:120:19: 'TaskPriority' is only available in iOS 13.0 or newer

/Users/******/Library/Developer/Xcode/DerivedData/PLAiOS-apjhsujbihydtdbingbknhgellax/SourcePackages/checkouts/replicate-swift/Sources/Replicate/Prediction.swift:137:19: 'TaskPriority' is only available in iOS 13.0 or newer



Build failed    11.12.2022 16:09    56.2 seconds
```

--- a/Package.resolved
+++ b/Package.resolved
@@ -14,7 +14,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/mattt/OpenAPIKit",
       "state" : {
-        "revision" : "86e2a2a762efae13b482ad64b2abe416a7c85cd8"
+        "revision" : "ebd4e93d22ccc0404b4e431f19987b65523849d2",
+        "version" : "2.5.1-fork.1"
       }
     },
     {

--- a/Package.swift
+++ b/Package.swift
@@ -21,7 +21,7 @@ let package = Package(
     dependencies: [
         // Dependencies declare other packages that this package depends on.
         .package(url: "https://github.com/Flight-School/AnyCodable", .upToNextMinor(from: "0.6.6")),
-        .package(url: "https://github.com/mattt/OpenAPIKit", revision: "86e2a2a762efae13b482ad64b2abe416a7c85cd8"),
+        .package(url: "https://github.com/mattt/OpenAPIKit", exact: "2.5.1-fork.1"),
         .package(url: "https://github.com/apple/swift-argument-parser", .upToNextMinor(from: "1.1.4")),
         .package(url: "https://github.com/apple/swift-syntax.git", exact: "0.50700.00"),
         .package(url: "https://github.com/apple/swift-format.git", exact: "0.50700.00"),

